### PR TITLE
CT-1671 AssignedBusinessUnitFilter back end

### PR DESCRIPTION
**Pull request by stephenrichards**

### CODE REVIEW

**Review by aliuk2012 — APPROVED** [review]
Looks good from frontend point of view.

File: app/assets/javascripts/modules/AssignedToFilter.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+moj.Modules.AssignedToFilter = {
+
+    init: function () {
+
+        if ($('#search_query_business_unit_name_filter').length === 0 ){
```
Comment (misaka) [inline app/assets/javascripts/modules/AssignedToFilter.js]: Why do we need this as on line 10 we only operate on `#search_query_business_unit_name_filter`?

File: app/assets/javascripts/modules/AssignedToFilter.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+
+
+    getMatchingBusinessUnits: function (searchFor){
+        $('.js-all-business-units').find('.multiple-choice')
+            .each(function( index, element ) {
+                var $elem = $(this);
+                var labelText = $elem.text().replace(/[^a-z0-9+]+/gi, '');
+                var cleanedTerm = searchFor.replace(/[^a-z0-9+]+/gi, '');
+                var check = new RegExp(cleanedTerm, 'gi')
```
Comment (misaka) [inline app/assets/javascripts/modules/AssignedToFilter.js]: I had to talk to you to understand this a bit better ... the use a `RegExp` here implies that there could be special characters in `check` that could confuse the results, although they're being stripped out so there's no real need to use a regular expression match anyway. Not a big issue, probably worth replacing this with a substring match as and when convenient.

**Review by misaka — APPROVED** [review]
Have made some comments on the front-end, but LGTM overall.

**Pull request merged**